ng-dev/auth/shared/node-handler.ts

```diff
@@ -0,0 +1,144 @@
+import {EventEmitter} from 'events';
+import * as Http from 'http';
+import * as Url from 'url';
+
+import {AuthorizationRequest} from '@openid/appauth';
+import {AuthorizationRequestHandler, AuthorizationRequestResponse} from '@openid/appauth';
+import {AuthorizationError, AuthorizationResponse} from '@openid/appauth';
+import {AuthorizationServiceConfiguration} from '@openid/appauth';
+import {Crypto} from '@openid/appauth';
+import {log} from '@openid/appauth';
+import {BasicQueryStringUtils, QueryStringUtils} from '@openid/appauth';
+import {NodeCrypto} from '@openid/appauth/built/node_support/crypto_utils.js';
+
+import opener from 'opener';
+
+class ServerEventsEmitter extends EventEmitter {
+  static ON_UNABLE_TO_START = 'unable_to_start';
+  static ON_AUTHORIZATION_RESPONSE = 'authorization_response';
+}
+
+export class NodeBasedHandler extends AuthorizationRequestHandler {
```

Comment:
Can you elaborate / and document in JSDoc, why we extend this and why we couldn't use the official/default request handler? e.g. `BaseTokenRequestHandler ` from https://github.com/openid/AppAuth-JS

Reply:
I added a TODO to move to it after their next release.  We just need a couple things to be fixed first.

Reply:
I see. yeah, this was not clear to me.